lms/djangoapps/dashboard/sysadmin.py

```diff
@@ -781,6 +784,96 @@ def get(self, request, *args, **kwargs):
         return render_to_response(self.template_name, context)
 
 
+class MgmtCommands(SysadminDashboardView):
+    """This class provides a view to handle the execution of management commands"""
+
+    def get(self, request):
+        """Displays the view for the management commands tab"""
+
+        if not request.user.is_staff:
+            raise Http404
+
+        context = {
+            'msg': self.msg,
+            'djangopid': os.getpid(),
+            'modeflag': {'mgmt_commands': 'active-section'},
+            'edx_platform_version': getattr(settings, 'EDX_PLATFORM_VERSION_STRING', ''),
+        }
+        return render_to_response(self.template_name, context)
+
+
+class ExecuteCommand(SysadminDashboardView):
+    """this class handles the execution of any management commands via ajax"""
+
+    def execute_command_and_capture_output(self, command, *args, **kwargs):
+        """
+        Executes the specified command with the given args and kwargs, and returns the console output
+        :param command:
+        :param args:
+        :param kwargs:
+        :return: client_stdout, client_stderr, client_error
+        """
+        orig_stdout = sys.stdout
+        orig_stderr = sys.stderr
+
+        sys.stdout = stdout_buffer = StringIO.StringIO()
+        sys.stderr = stderr_buffer = StringIO.StringIO()
+
+        client_error = None
+        try:
+            call_command(command, *args, **kwargs)
+        except SystemExit:
```

Comment:
Good question, 
To see when this happens, select the management command, 'Whitelist a user': and click execute command with no options.

This is what can cause this:

Some management commands raise a CommandError. In order to execute a command, 
we use django.core.management.call_command. call_command first loads the default arguments for the command, and then calls klass.execute, which calls the "handle" function of the particular management command. when calling the particular "handle" method, klass.execute catches all CommandError exceptions and calls  sys.exit(1). Therefore, instead of catching CommandError exceptions, we catch SystemExit exception. Let me know if that does not make sense.